### ISSUE TITLE
No Imported Games

### ISSUE THREAD

**Issue by lordfiSh** [UncleGoogle/galaxy-integration-humblebundle#31]
**Describe the bug**
Games from the Humble Library are not imported

**Expected behavior**
Games from the Humble Library are imported

**Screenshots**
![image](https://user-images.githubusercontent.com/389680/65706381-a6334f80-e08a-11e9-9364-767718f462ea.png)
![image](https://user-images.githubusercontent.com/389680/65706393-ae8b8a80-e08a-11e9-850d-33570c896943.png)

**Attach logs**
[plugin-humble-f0ca3d80-a432-4d35-a9e3-60f27161ac3a.zip](https://github.com/UncleGoogle/galaxy-integration-humblebundle/files/3658834/plugin-humble-f0ca3d80-a432-4d35-a9e3-60f27161ac3a.zip)

**Comment by japzone1** [comment]
I'm also having this issue. Not seeing any errors, and I successfully logged into my account, but my Humble Bundle game list is just empty.

**Comment by UncleGoogle** [comment]
Hi, thanks for report. Please try this:
[humblebundle_0.4.1_parsefix.zip](https://github.com/UncleGoogle/galaxy-integration-humblebundle/files/3665646/humblebundle_0.4.1_parsefix.zip)

**Comment by lordfiSh** [comment]
Working now, but some Games can't be imported. 

![image](https://user-images.githubusercontent.com/389680/65822720-17b8fc80-e249-11e9-953f-e6972f5c6e6f.png)

[plugin-humble-f0ca3d80-a432-4d35-a9e3-60f27161ac3a.zip](https://github.com/UncleGoogle/galaxy-integration-humblebundle/files/3666035/plugin-humble-f0ca3d80-a432-4d35-a9e3-60f27161ac3a.zip)

**Comment by UncleGoogle** [comment]
Uh, looks bad. Those Unknown Games is gog bug: see #28 

Regarding to the original issue: can you send your logs to me? There was a crash on one game that have no "human_name". I haven't seen such case before, so I'm interested what type of product is this.

**Comment by lordfiSh** [comment]
What Logfile?

**Comment by UncleGoogle** [comment]
All rolled logs (plugin-humble-xxx.log#)

**Comment by lordfiSh** [comment]
[logs.zip](https://github.com/UncleGoogle/galaxy-integration-humblebundle/files/3666519/logs.zip)

**Comment by UncleGoogle** [comment]
Thanks! 
Hah, fun fact: 
- I set up sentry to be integrated with main logger and send me any WARNING and ERROR types.
- Your DNS Blocker block sentry to send reports
- Galaxy API was set to log all unsuccessful connections as warnings; there was 3 retries for every connection by default in urllib3
- sentry was trying to send reports about all those warning logs...
- and so on. In few minutes you have 100MB of trash log, totally nothing interesting inside.

I've changed logging to error level. If you wish to help, you can either use this and paste your logs here, or stop blocking sentry for a while, then restart Galaxy - I should receive all information.
[humblebundle_0.4.1_sentryfix.zip](https://github.com/UncleGoogle/galaxy-integration-humblebundle/files/3667597/humblebundle_0.4.1_sentryfix.zip)

**Comment by lordfiSh** [comment]
I've whitelisted the Domain and seen a few Connections, see if see any logs now.

**Comment by UncleGoogle** [comment]
Yes, I got single steam key without "human_name":
```
{
disallowed_countries: [], 
display_separately: False, 
gamekey: mVsbh4Fsvy8S, 
instructions_html: <a href='https://support.humblebundle.com/hc/en-us/articles/204008710-How-To-Redeem-Steam-Keys' target='_blank'>Steam Instructions</a>, 
is_gift: False, 
key_type: steam, 
keyindex: 0, 
machine_name: steamandroidbundle, 
redeemed_key_val: ***, 
show_custom_instructions_in_user_libraries: False, 
visible: True
}

Looks not important for now... And it looks like a bunch of games under single key, right? How it is presented in hb library?
```

**Comment by UncleGoogle** [comment]
Can you send me an email with output of below link when you are logged in humble in browser: https://www.humblebundle.com/api/v1/order/mVsbh4Fsvy8S?all_tpkds=true
? Remove first all sensitive data ("redeemed_key_val" if you revealed but not redeemed)

**Comment by lordfiSh** [comment]
send

**Comment by lordfiSh** [comment]
seems to be fixed, no more Unknown Games: https://mantis2.gog.com/view.php?id=13136

**Comment by UncleGoogle** [comment]
epilog.

Problem was no "human_name" in tpkd structure (third party key) returned by humblebundle API. It is important as without human name GOG ppl cannot guess and match game_id with proper title.
And people also cannot identify such game key. lordfiSh sent me screenshot:

![image (2)](https://user-images.githubusercontent.com/12197825/67719181-9ed8cc00-f9d1-11e9-8b8c-a0fa99d6d1f1.png)
(the upper key has no human-friendly name)

Fix from my side was to ignore such key when there is no human_name to avoid crashes.

I've wrote to humblebundle support and they fix this game case:
>You should now see the name "EDGE, Anomaly, and Osmos Steam Key" as our operations team was able to resolve the issue at hand

>In certain cases, these products may be set up by developers, and they may have not submitted a human-friendly name with the build, though I cannot say for sure in this case.

Then I've found and report in hb support another 6 keys w/o human_name and will do the same if meet more.
